python/ray/data/_internal/stats.py

```diff
@@ -419,16 +422,18 @@ def to_string(
             out += "* Extra metrics: " + str(self.extra_metrics) + "\n"
         out += str(self.iter_stats)
 
-        mb_spilled = round(self.global_bytes_spilled / 1e6)
-        mb_restored = round(self.global_bytes_restored / 1e6)
-        if (
-            len(self.stages_stats) > 0
-            and add_global_stats
-            and (mb_spilled or mb_restored)
-        ):
-            out += "\nCluster memory:\n"
-            out += "* Spilled to disk: {}MB\n".format(mb_spilled)
-            out += "* Restored from disk: {}MB\n".format(mb_restored)
+        if len(self.stages_stats) > 0 and add_global_stats:
+            mb_spilled = round(self.global_bytes_spilled / 1e6)
+            mb_restored = round(self.global_bytes_restored / 1e6)
+            if mb_spilled or mb_restored:
+                out += "\nCluster memory:\n"
+                out += "* Spilled to disk: {}MB\n".format(mb_spilled)
+                out += "* Restored from disk: {}MB\n".format(mb_restored)
+
+            dataset_mb_spilled = round(self.dataset_bytes_spilled / 1e6)
+            if dataset_mb_spilled:
+                out += "\nDataset memory:\n"
+                out += "* Spilled to disk: {}MB\n".format(dataset_mb_spilled)
```

Comment:
if we want to always log the dataset spilled bytes, should we leave this outside the add_global_stats check?

Reply:
let's also include `add_global_stats` in the function docstring

Reply:
`add_global_stats` means that we'll only log this for this `DatasetStatSummary`, and none of it's parents. Since this is a dataset-level stat, it'll be the same for all `DatasetStatSummary`s.